Add ccagt dataset (#4486)
huggingface/datasets: datasets/ccagt/README.md

---
annotations_creators:
- expert-generated
language_creators:
- expert-generated
languages:
- en
licenses:
- cc-by-nc-3.0
multilinguality:
- monolingual
paperswithcode_id: null
pretty_name: Images of Cervical Cells with AgNOR Stain Technique
size_categories:
- 1K<n<10K
source_datasets:
- original
task_categories:
- image-segmentation
- object-detection
task_ids:
- semantic-segmentation
---

# Dataset Card for Images of Cervical Cells with AgNOR Stain Technique

> ## Table of Contents
> - [Dataset Description](#dataset-description)
>   - [Dataset Summary](#dataset-summary)
>   - [Supported Tasks](#supported-tasks-and-leaderboards)
>   - [Languages](#languages)
> - [Dataset Structure](#dataset-structure)
>   - [Data Instances](#data-instances)
>   - [Data Fields](#data-instances)
>   - [Data Splits](#data-instances)
> - [Dataset Creation](#dataset-creation)
>   - [Curation Rationale](#curation-rationale)
>   - [Source Data](#source-data)
>   - [Annotations](#annotations)
>   - [Personal and Sensitive Information](#personal-and-sensitive-information)
> - [Considerations for Using the Data](#considerations-for-using-the-data)
>   - [Social Impact of Dataset](#social-impact-of-dataset)
>   - [Other Known Limitations](#other-known-limitations)
> - [Additional Information](#additional-information)
>   - [Dataset Curators](#dataset-curators)
>   - [Licensing Information](#licensing-information)
>   - [Citation Information](#citation-information)

Reviewer: ```suggestion
## Table of Contents
- [Dataset Description](#dataset-description)
  - [Dataset Summary](#dataset-summary)
  - [Supported Tasks and Leaderboards](#supported-tasks-and-leaderboards)
  - [Languages](#languages)
- [Dataset Structure](#dataset-structure)
  - [Data Instances](#data-instances)
  - [Data Fields](#data-fields)
  - [Data Splits](#data-splits)
- [Dataset Creation](#dataset-creation)
  - [Curation Rationale](#curation-rationale)
  - [Source Data](#source-data)
  - [Annotations](#annotations)
  - [Personal and Sensitive Information](#personal-and-sensitive-information)
- [Considerations for Using the Data](#considerations-for-using-the-data)
  - [Social Impact of Dataset](#social-impact-of-dataset)
  - [Discussion of Biases](#discussion-of-biases)
  - [Other Known Limitations](#other-known-limitations)
- [Additional Information](#additional-information)
  - [Dataset Curators](#dataset-curators)
  - [Licensing Information](#licensing-information)
  - [Citation Information](#citation-information)
  - [Contributions](#contributions)
```

- `annotation`: A `PIL.Image.Image` object containing the annotation mask.

#### `object detection`

- `image`: A `PIL.Image.Image` object containing the image. Note that when accessing the image column: `dataset[0]["image"]` the image file is automatically decoded. Decoding of a large number of image files might take a significant amount of time. Thus it is important to first query the sample index before the `"image"` column, *i.e.* `dataset[0]["image"]` should **always** be preferred over `dataset["image"][0]`.
- `objects`: a dictionary containing bounding boxes and labels of the cell objects 
  - `bbox`: a list of bounding boxes (in the [coco](https://albumentations.ai/docs/getting_started/bounding_boxes_augmentation/#coco) format) corresponding to the objects present on the image

>   - `label`: a list of integers representing the category (7 categories to describe the objects in total; two to differentiate nucleolus organizer regions), with the possible values including `NUCLEUS` (1), `CLUSTER` (2), `SATELLITE` (3), `NUCLEUS_OUT_OF_FOCUS` (4), `OVERLAPPED_NUCLEI` (5), `NON_VIABLE_NUCLEUS` (6) and `LEUKOCYTE_NUCLEUS` (7).

Reviewer: `datasets` / `transformers` expect the `0, ..., num_classes-1` label range

```suggestion
  - `label`: a list of integers representing the category (7 categories to describe the objects in total; two to differentiate nucleolus organizer regions), with the possible values including `NUCLEUS` (0), `CLUSTER` (1), `SATELLITE` (2), `NUCLEUS_OUT_OF_FOCUS` (3), `OVERLAPPED_NUCLEI` (4), `NON_VIABLE_NUCLEUS` (5) and `LEUKOCYTE_NUCLEUS` (6).
```

Author: I think it makes more sense to keep the same order/IDs as in semantic segmentation, where 0 is for background/ignored pixels

#### Initial Data Collection and Normalization

The images are collected as patches/tiles of whole slide images (WSIs) from cervical samples stained with AgNOR technique to allow the detection of nucleolus organizer regions (NORs). NORs are DNA loops containing genes responsible for the transcription of ribosomal RNA located in the cell nucleolus. They contain a set of argyrophilic proteins, selectively stained by silver nitrate, which can be identified as black dots located throughout the nucleoli area and called AgNORs.

#### Who are the source language producers?

The dataset was built using images from examinations (a gynecological exam, colposcopy and biopsy) of 15 women patients who were treated at the Gynecology and Colposcopy Outpatient Clinic of the [University Hospital Professor Polydoro Ernani de São Thiago of Federal University of Santa Catarina (HU-UFSC)](https://unihospital.ufsc.br/) and had 6 different diagnoses in their oncological exams. The samples were collected by the members of the Clinical Analyses Department: Ane Francyne Costa, Fabiana Botelho De Miranda Onofre, and Alexandre Sherlley Casimiro Onofre.

### Annotations

#### Annotation process

> The instances were annotated using the [labelbox](https://labelbox.com/) tool. The satellite category was labeled as a single dot, and the other categories were labeled as polygons. After the annotation process, all annotations was reviewed.

Reviewer: ```suggestion
The instances were annotated using the [labelbox](https://labelbox.com/) tool. The satellite category was labeled as a single dot, and the other categories were labeled as polygons. After the annotation process, all annotations were reviewed.
```

- `image`: A `PIL.Image.Image` object containing the image. Note that when accessing the image column: `dataset[0]["image"]` the image file is automatically decoded. Decoding of a large number of image files might take a significant amount of time. Thus it is important to first query the sample index before the `"image"` column, *i.e.* `dataset[0]["image"]` should **always** be preferred over `dataset["image"][0]`.
- `objects`: a dictionary containing bounding boxes and labels of the cell objects
  - `bbox`: a list of bounding boxes (in the [coco](https://albumentations.ai/docs/getting_started/bounding_boxes_augmentation/#coco) format) corresponding to the objects present on the image
  - `segment`: a list of segments in format of `[polygon_0, ..., polygon_n]`, where each polygon is `[x0, y0, ..., xn, yn]`.
  - `label`: a list of integers representing the category (7 categories to describe the objects in total; two to differentiate nucleolus organizer regions), with the possible values including `NUCLEUS` (0), `CLUSTER` (1), `SATELLITE` (2), `NUCLEUS_OUT_OF_FOCUS` (3), `OVERLAPPED_NUCLEI` (4), `NON_VIABLE_NUCLEUS` (5) and `LEUKOCYTE_NUCLEUS` (6).

### Data Splits

> The data is split randomly using the fixed seed into training, test and validation set. The training data contains 70% of the images and the testing and the validation data contain 15% of the images each. In total, the training set contains 6533 images and the testing and the validation set 1403 images each.

Reviewer 2: Are the splits provided by the dataset authors ? Or is this an arbitrary splitting that you implemented yourself ?

We usually only use official splits, or we let the users define their own splits

Author: I am the author of the dataset, so yes it's the official splitting process 😄 

The difference to the official results (at [10.2139/ssrn.4126881](https://dx.doi.org/10.2139/ssrn.4126881)), was that in the paper we also added all the data from one slide to the test set and then performed this split... i can also add this slide to the test fold to be equals, but I would like to provide this to the users so that they can choose which slide they would like to separate at the start

  - [Citation Information](#citation-information)
  - [Contributions](#contributions)

## Dataset Description

- **Homepage:** [CCAgT homepage](https://data.mendeley.com/datasets/wg4bpm33hj/)
- **Repository:** [CCAgT-utils](https://doi.org/10.5281/zenodo.6350442)
- **Paper:** [Semantic Segmentation for the Detection of Very Small Objects on Cervical Cell Samples Stained with the AgNOR Technique](https://dx.doi.org/10.2139/ssrn.4126881)
- **Leaderboard:** [Needs More Information]
- **Point of Contact:** [João G. A. Amorim](mailto:joao.atkinson@posgrad.ufsc.br)

### Dataset Summary

The CCAgT (Images of Cervical Cells with AgNOR Stain Technique) dataset contains 9339 images (1600x1200 resolution where each pixel is 0.111µmX0.111µm) from 15 different slides stained using the AgNOR technique. Each image has at least one label. In total, this dataset has more than 63K instances of annotated object. The images are from the patients of the Gynecology and Colonoscopy Outpatient Clinic of the [Polydoro Ernani de São Thiago University Hospital of the Universidade Federal de Santa Catarina (HU-UFSC)](https://unihospital.ufsc.br/).

### Supported Tasks and Leaderboards

> - `image-segmentation`: The dataset can be used to train a model for semantic segmentation or instance segmentation. Semantic segmentation consists in classifying each pixel of the image. Instance segmentation consists of doing object detection first and then using a semantic segmentation model inside detected objects. Success on this task is typically measured by achieving high values of [mean iou](https://huggingface.co/spaces/evaluate-metric/mean_iou) or [f-score](https://huggingface.co/spaces/evaluate-metric/f1) for pixels results. For instances results, this task is typically measured by achieving high values of [recall](https://huggingface.co/spaces/evaluate-metric/recall), [precision](https://huggingface.co/spaces/evaluate-metric/precision) and [f-score](https://huggingface.co/spaces/evaluate-metric/f1).
>   - Panoptic segmentation require an additional conversion of the original dataset. Panoptic segmentation is the combination of instance segmentation and semantic segmentation.

Reviewer: Let's reorder the sentences here and remove the panoptic segmentation part to simplify the task entry:

```suggestion
- `image-segmentation`: The dataset can be used to train a model for semantic segmentation or instance segmentation. Semantic segmentation consists in classifying each pixel of the image. Success on this task is typically measured by achieving high values of [mean iou](https://huggingface.co/spaces/evaluate-metric/mean_iou) or [f-score](https://huggingface.co/spaces/evaluate-metric/f1) for pixels results. Instance segmentation consists of doing object detection first and then using a semantic segmentation model inside detected objects. For instances results, this task is typically measured by achieving high values of [recall](https://huggingface.co/spaces/evaluate-metric/recall), [precision](https://huggingface.co/spaces/evaluate-metric/precision) and [f-score](https://huggingface.co/spaces/evaluate-metric/f1).
```

- `object-detection`: The dataset can be used to train a model for object detection to detect the nuclei categories or the nucleolus organizer regions (NORs), which consists of locating instances of objects and then classifying each one. This task is typically measured by achieving a high values of [recall](https://huggingface.co/spaces/evaluate-metric/recall), [precision](https://huggingface.co/spaces/evaluate-metric/precision) and [f-score](https://huggingface.co/spaces/evaluate-metric/f1).

### Languages

The class labels in the dataset are in English.

## Dataset Structure

### Data Instances

An example looks like the one below:

#### `semantic segmentation` (default configuration)

```
{
  'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=1200x1600 at 0x276021C5EB8>,
  'annotation': <PIL.PngImagePlugin.PngImageFile image mode=L size=1200x1600 at 0x385021C5ED7>
}
```

#### `object detection`

```
{
  'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=1200x1600 at 0x276021C5EB8>,
  'objects': {
    'bbox': [
      [36, 7, 13, 32],
      [50, 7, 12, 32]
    ], 
    'label': [1, 5]
  }
```

> 

Reviewer: Let's add an example for the `instance segmentation`  config here.

### Data Fields

The data annotations have the following fields:

#### `semantic segmentation` (default configuration)

- `image`: A `PIL.Image.Image` object containing the image. Note that when accessing the image column: `dataset[0]["image"]` the image file is automatically decoded. Decoding of a large number of image files might take a significant amount of time. Thus it is important to first query the sample index before the `"image"` column, *i.e.* `dataset[0]["image"]` should **always** be preferred over `dataset["image"][0]`.

> - `annotation`: A `PIL.Image.Image` object containing the annotation mask. The mask has the category id values, with the possible values including `BACKGROUND` (0), `NUCLEUS` (1), `CLUSTER` (2), `SATELLITE` (3), `NUCLEUS_OUT_OF_FOCUS` (4), `OVERLAPPED_NUCLEI` (5), `NON_VIABLE_NUCLEUS` (6) and `LEUKOCYTE_NUCLEUS` (7).

Reviewer: ```suggestion
- `annotation`: A `PIL.Image.Image` object containing the annotation mask. The mask has a single channel and the following pixel values are possible: `BACKGROUND` (0), `NUCLEUS` (1), `CLUSTER` (2), `SATELLITE` (3), `NUCLEUS_OUT_OF_FOCUS` (4), `OVERLAPPED_NUCLEI` (5), `NON_VIABLE_NUCLEUS` (6) and `LEUKOCYTE_NUCLEUS` (7).
```
